src/pipeline/Pipeline.cpp

```diff
@@ -657,6 +661,22 @@ CalibrationHandler PipelineImpl::getCalibrationData() const {
     }
 }
 
+void PipelineImpl::setEepromData(EepromData eepromData) {
+    globalProperties.calibData = eepromData;
+}
+
+bool PipelineImpl::isEepromDataAvailable() const {
+    return globalProperties.calibData.has_value();
+}
+
+EepromData PipelineImpl::getEepromData() const {
+    if(globalProperties.calibData) {
+        return globalProperties.calibData.value();
+    } else {
+        return EepromData{};
+    }
+}
```

Comment:
```suggestion
tl::optional<EepromData> PipelineImpl::getEepromData() const {
    return globalProperties.calibData;
}
```

For this one, as we don't have to retain API